src/gtk/webview_webkit2.cpp

```diff
@@ -608,6 +682,80 @@ wxgtk_authorize_authenticated_peer_cb(GDBusAuthObserver *,
 
 } // extern "C"
 
+//-----------------------------------------------------------------------------
+// wxWebViewConfigurationImplWebKit
+//-----------------------------------------------------------------------------
+class wxWebViewConfigurationImplWebKit : public wxWebViewConfigurationImpl
+{
+public:
+    wxWebViewConfigurationImplWebKit():
+        m_webContext(nullptr)
+#if WEBKIT_CHECK_VERSION(2, 10, 0)
+        ,
+        m_websiteDataManager(nullptr)
+#endif
+    { }
+
+#if WEBKIT_CHECK_VERSION(2, 10, 0)
+    wxString GetDataPath() const override
+    {
+        GetOrCreateContext();
+        if (m_websiteDataManager)
+            return webkit_website_data_manager_get_base_data_directory(m_websiteDataManager);
```

Comment:
We probably need a run-time version check for 2.10 here (and when creating it below), otherwise an app compiled on a newer system will crash when running on an older one and I think we want to support this.

As I've just recently written in [this comment](https://github.com/wxWidgets/wxWidgets/issues/23367#issuecomment-1479377777), I think it would be useful to add `wx_check_webkit_version()` function for doing it, as WebKit itself doesn't seem to provide one.

Reply:
Yes, it's called "lazy binding" and is the default under Linux (and the vast majority of the other Unix systems although historically there were some exceptions, I believe). This is also the default behaviour of `dlopen()` (`RTLD_LAZY` is the default, but `RTLD_NOW` can be used to force immediately resolving everything).